homeassistant/components/tellduslive/entry.py

```diff
@@ -0,0 +1,114 @@
+"""Base Entity for all TelldusLiveEntities."""
+from datetime import datetime
+import logging
+
+from homeassistant.const import ATTR_BATTERY_LEVEL, DEVICE_DEFAULT_NAME
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+from .const import SIGNAL_UPDATE_ENTITY
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_LAST_UPDATED = 'time_last_updated'
+
+
+class TelldusLiveEntity(Entity):
+    """Base class for all Telldus Live entities."""
+
+    def __init__(self, client, device_id):
+        """Initialize the entity."""
+        self._id = device_id
+        self._client = client
+        self._name = self.device.name
+        self._async_unsub_dispatcher_connect = None
+
+    async def async_added_to_hass(self):
+        """Call when entity is added to hass."""
+        _LOGGER.debug('Created device %s', self)
+        self._async_unsub_dispatcher_connect = async_dispatcher_connect(
+            self.hass, SIGNAL_UPDATE_ENTITY, self._update_callback)
+        self._update_callback()
```

Comment:
This will add an extra state update, that looks not needed. State will be updated as the last step of entity addition.